presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -86,6 +87,7 @@
     private final LoadingCache<String, List<String>> databaseNamesCache;
     private final LoadingCache<HiveTableName, Optional<Table>> tableCache;
     private final LoadingCache<String, List<String>> tableNamesCache;
+    private final LoadingCache<List<String>, List<String>> tablesWithParameterCache;
```

Comment:
I'd rather create a different class for `<databaseName, parameterKey, parameterValue>` than using a list to key the cache.

Reply:
nit: we can name it something like`tableNamesCacheForPattern` for consistency with `tableNamesCache`. both of them fetch tables for a database, the later on takes a filter into account.

Reply:
> nit: we can name it something like`tableNamesCacheForPattern` for consistency with `tableNamesCache`. both of them fetch tables for a database, the later one takes a filter into account.

"Pattern" might not be very accurate because we use both `=` and `LIKE` when constructing the filter and we restrict the filter value to be alphanumeric. And I think it should be better if we just name the cache field after the method whose results it is caching.